synapse/app/homeserver.py

```diff
@@ -132,8 +132,10 @@ def _listener_http(
         self._module_web_resources_consumed = True
 
         # try to find something useful to redirect '/' to
-        if WEB_CLIENT_PREFIX in resources:
-            root_resource: Resource = RootOptionsRedirectResource(WEB_CLIENT_PREFIX)
+        if self.config.server.web_client_location_is_redirect:
```

Comment:
Can you spell out why these two are equivalent? I'm not following (but not sure how the `resources` stuff works.

Reply:
They are not equivalent, as i mentioned in the description:

> if `web_client_location` points to an HTTP(S) URL than `/` will now redirect to it, regardless of whether `webclient` is a configured listener.

Reply:
I *think* this is acceptable and what we're going for, but I'm not 100% sure. 👍